SpatialGDK/Source/SpatialGDK/Public/Schema/UnrealObjectRef.cpp

```diff
@@ -2,5 +2,128 @@
 
 #include "Schema/UnrealObjectRef.h"
 
+#include "EngineClasses/SpatialPackageMapClient.h"
+#include "Utils/SchemaUtils.h"
+
+DEFINE_LOG_CATEGORY_STATIC(LogUnrealObjectRef, Log, All);
+
 const FUnrealObjectRef FUnrealObjectRef::NULL_OBJECT_REF = FUnrealObjectRef(0, 0);
 const FUnrealObjectRef FUnrealObjectRef::UNRESOLVED_OBJECT_REF = FUnrealObjectRef(0, 1);
+
+UObject* FUnrealObjectRef::ToObjectPtr(const FUnrealObjectRef& ObjectRef, USpatialPackageMapClient* PackageMap, bool& bOutUnresolved)
+{
+	if (ObjectRef == FUnrealObjectRef::NULL_OBJECT_REF)
+	{
+		return nullptr;
+	}
+	else
+	{
+		if (ObjectRef.bSingletonRef)
+		{
+			FUnrealObjectRef SingletonClassRef = ObjectRef;
+			SingletonClassRef.bSingletonRef = false;
```

Comment:
Is it some trick? :) Setting SingletonClassRef.bSingletonRef = false for an object that is a singleton ref?

Reply:
As discussed - this value SingletonClassRef.bSingletonRef is set to false here to make GetTypeHash function (in header) future proof (if anyone decides to add bSingletonRef to it - Find function for a map will break).

Reply:
So I find it somewhat misleading and would prefer to not have this assignment, but rather have a comment in GetTypeHash function. Also, the variable bSingletonRef could be renamed to use_singleton_class_path  as already mentioned in one of the comments